tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7708,6 +7708,10 @@ def err_hlsl_varmodifierna_decltype : Error<
   "%0 is not a valid modifier for a declaration of type %1">;
 def note_hlsl_globallycoherent_applies_to : Note<
   "'globallycoherent' can only be applied to UAV or RWDispatchNodeInputRecord objects">;
+def note_hlsl_reordercoherent_applies_to : Note<
```

Comment:
It would be nice to combine this with the `note_hlsl_globallycoherent_applies_to` text. The annotation keyword should be something we can provide as an input since we should have the attribute available, and the work graphs type could be under a `select`.

The reason these matter is two-fold, one it condenses the constant string data just a little bit, but more importantly there are hardcoded limits to the number of diagnostics we can have, and each time we increase that limit it comes with a memory impact.

Reply:
Fused the two diags into one